Mathlib/Tactic/Rewrites.lean

```diff
@@ -166,11 +167,14 @@ def rewrites (lemmas : DiscrTree (Name × Bool × Nat) s × DiscrTree (Name × B
     (goal : MVarId) (target : Expr) (stop_at_rfl : Bool := False) (max : Nat := 20)
     (leavePercentHeartbeats : Nat := 10) : MetaM (List RewriteResult) := do
   let results ← rewritesCore lemmas goal target
-    -- Don't use too many heartbeats.
-    |>.whileAtLeastHeartbeatsPercent leavePercentHeartbeats
+    -- Don't report duplicate results.
+    -- (TODO: we later pretty print results; save them here?)
+    |>.dedupBy (fun r => do pure <| (← ppExpr r.result.eNew).pretty)
```

Comment:
Do I understand correctly that this could remove results that are actually different goals if they pretty print the same? For instance if they result in "the same goal" but where coercion arrows actually put everything in a different type?
I think this is fine as a first pass but maybe we can add a TODO note here or as an issue that maybe some more refined logic here would be an improvement